builder/azure/arm/step_deploy_template.go

```diff
@@ -298,21 +286,48 @@ func (s *StepDeployTemplate) deleteDeploymentResources(ctx context.Context, subs
 	}
 
 	resources := map[string]string{}
-
+	var vmID *virtualmachines.VirtualMachineId
+	var nicID *commonids.NetworkInterfaceId
 	for _, deploymentOperation := range deploymentOperations.Items {
 		// Sometimes an empty operation is added to the list by Azure
 		if deploymentOperation.Properties.TargetResource == nil {
 			continue
 		}
-
 		resourceName := *deploymentOperation.Properties.TargetResource.ResourceName
 		resourceType := *deploymentOperation.Properties.TargetResource.ResourceType
 
-		s.say(fmt.Sprintf("Adding to deletion queue -> %s : '%s'", resourceType, resourceName))
-		resources[resourceType] = resourceName
-
+		// Create vm and NIC ID to use them to delete later
```

Comment:
One small quirk of this change is that previously we would call Delete VM/NIC even if it failed once, we don't expect this to happen but technically we would keep retrying that call. So far I've only seen the Public IP and VNET deletions would fail due to a still existing network interface. So it shouldn't be a problem but just figured I'd call it out and see if the reviewer of this PR has any thoughts :D

Reply:
One option is just to ship it as is, and then if we here reports from users that they're seeing orphaned VMs we can quickly re-add it to the deletion queue in the event that it fails, basically falling back on the old mechanism

Reply:
We now retry this